guides/advanced_guides/datatypes.md

```diff
@@ -0,0 +1,135 @@
+# Data Types
+
+This guide describes the different data types supported for the fields in a document and how MeiliSearch handles them.
+
+No matter the type, the value of a field will remain unchanged in the returned documents upon search.
+For example, if you have a complex field with nested objects, this field will be returned with the same complexity upon search.
+
+Based on their type, however, the fields will be handled and used in different ways by MeiliSearch. **The type affects how a field is used for search results**.
+
+Types:
+
+- [String](/guides/advanced_guides/datatypes.md#string)
+- [Numeric types](/guides/advanced_guides/datatypes.md#numeric-types): `integer`, `float`
+- [Boolean](/guides/advanced_guides/datatypes.md#boolean)
+- [Array](/guides/advanced_guides/datatypes.md#array)
+- [Object](/guides/advanced_guides/datatypes.md#object)
+- [`null` type](/guides/advanced_guides/datatypes.md#null-type)
+
+### String
+
+String is the primary type for indexing data in MeiliSearch. It enables to create the content in which to search. Strings are processed as detailed below.
+
+> String tokenization is the process of splitting a string into a list of individual terms that are called tokens.
+
+A string is passed to a tokenizer and is then broken into separate string tokens. A token is a **word**.
+
+- For Latin-based languages, the words are separated by **space**.
+- For Kanji characters, the words are separated by **character**.
+
+For Latin-based languages, there are two kinds of **space separators**:
+
+- **Soft spaces** (distance: 1): whitespaces, quotes, `'-' | '_' | '\'' | ':' | '/' | '\\' | '@'`
+- **Hard spaces** (distance: 8): `'.' | ';' | ',' | '!' | '?' | '(' | ')'`
+
+Distance plays an essential role in in determining whether documents are relevant since [one of the ranking rules is the **proximity** rule](/guides/main_concepts/relevancy.md). The proximity rule sorts the results by increasing distance between matched query terms. Then, two words separated by a soft space are closer and thus considered **more relevant** than two words separated by a hard space.
```

Comment:
```suggestion
Distance plays an essential role in determining whether documents are relevant since [one of the ranking rules is the **proximity** rule](/guides/main_concepts/relevancy.md). The proximity rule sorts the results by increasing distance between matched query terms. Then, two words separated by a soft space are closer and thus considered **more relevant** than two words separated by a hard space.
```